test/MediatorTest.php

```diff
@@ -81,34 +81,4 @@ public function testMultiSubscriberErrorCase() {
         $this->assertTrue($caughtExpected);
         $this->assertSame(2, $invocations);
     }
-
-    public function testSubscriberArrayRemovedWhenEmptyToAvoidLeakage() {
```

Comment:
There's no reason to remove the tests. You are in possession of the class code, so you may as well just ignore the encapsulation here and the read private property here. (`$events = (function() { return $this->eventSubscriberMap; })->call($mediator);`)

Because that's essentially what you want to test: whether the internal state is correct.

Reply:
Good call. Updated 👍